### PR TITLE
Handle SYS_NET_IPPROTO_IP get/setsockopt

### PR DESCRIPTION
**Pull request by clienthax**
Improve net logging.

Should allow latest vsh to load (Depending on windows version apparently..)
Seems to work on 19043.1415, but not on 22000.348 - https://docs.microsoft.com/en-us/windows/win32/winsock/using-so-reuseaddr-and-so-exclusiveaddruse ?

```
[0x0001b1e0] sys_net: sys_net_bnet_socket(family=INET, type=DGRAM, protocol=IPPROTO_IP)
[0x0001b3c4] sys_net: sys_net_bnet_setsockopt(s=54, level=SYS_NET_IPPROTO_IP, optname=IP_TTL, optval=*0xd00b1130, optlen=4)
[0x0001b3c4] sys_net: optval: 0x00000001
[0x0001b3c4] sys_net: sys_net_bnet_setsockopt(s=54, level=SYS_NET_SOL_SOCKET, optname=SO_RCVTIMEO, optval=*0xd00b1138, optlen=16)
[0x0001be74] sys_net: sys_net_bnet_bind(s=54, addr=*0x13e0c64, addrlen=16)
[0x0001be74] sys_net: Trying to bind 0.0.0.0:9309

[0x0001b1e0] sys_net: sys_net_bnet_socket(family=INET, type=STREAM, protocol=IPPROTO_IP)
[0x0001b3c4] sys_net: sys_net_bnet_setsockopt(s=55, level=SYS_NET_SOL_SOCKET, optname=SO_REUSEADDR, optval=*0xd00b1140, optlen=4)
[0x0001b3c4] sys_net: optval: 0x00000001
[0x0001be74] sys_net: sys_net_bnet_bind(s=55, addr=*0xd00b1144, addrlen=16)
[0x0001be74] sys_net: Trying to bind 0.0.0.0:9309
[0x0001be74] sys_net: Socket error EACCES
[0x0001be74] SYS: 'sys_net_bnet_bind' failed with 0xfffffff3 : -SYS_NET_EACCES [1]
```

### CODE REVIEW

File: rpcs3/Emu/Cell/lv2/sys_net.cpp [hunk 1]
```diff
@@ -1980,7 +2073,20 @@ error_code sys_net_bnet_getsockopt(ppu_thread& ppu, s32 s, s32 level, s32 optnam
...
+		sys_net.warning("sys_net_bnet_getsockopt(s=%d, level=SYS_NET_SOL_SOCKET, optname=%s, optval=*0x%x, optlen=%u)", s, static_cast<lv2_socket_option>(optname), optval, optlen);
+		break;
+	case SYS_NET_IPPROTO_TCP:
+		sys_net.warning("sys_net_bnet_getsockopt(s=%d, level=SYS_NET_IPPROTO_TCP, optname=%s, optval=*0x%x, optlen=%u)", s, static_cast<lv2_tcp_option>(optname), optval, optlen);
+		break;
+	case SYS_NET_IPPROTO_IP:
+		sys_net.warning("sys_net_bnet_getsockopt(s=%d, level=SYS_NET_IPPROTO_IP, optname=%s, optval=*0x%x, optlen=%u)", s, static_cast<lv2_ip_option>(optname), optval, optlen);
+		break;
+	default:
+		sys_net.warning("sys_net_bnet_getsockopt(s=%d, level=0x%x, optname=0x%x, optval=*0x%x, optlen=%u)", s, level, optname, optval, optlen);
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/lv2/sys_net.cpp:2088]: Missing break

File: rpcs3/Emu/Cell/lv2/sys_net.cpp [hunk 2]
```diff
@@ -2827,7 +3004,21 @@ error_code sys_net_bnet_setsockopt(ppu_thread& ppu, s32 s, s32 level, s32 optnam
...
+		sys_net.warning("sys_net_bnet_setsockopt(s=%d, level=SYS_NET_SOL_SOCKET, optname=%s, optval=*0x%x, optlen=%u)", s, static_cast<lv2_socket_option>(optname), optval, optlen);
+		break;
+	case SYS_NET_IPPROTO_TCP:
+		sys_net.warning("sys_net_bnet_setsockopt(s=%d, level=SYS_NET_IPPROTO_TCP, optname=%s, optval=*0x%x, optlen=%u)", s, static_cast<lv2_tcp_option>(optname), optval, optlen);
+		break;
+	case SYS_NET_IPPROTO_IP:
+		sys_net.warning("sys_net_bnet_setsockopt(s=%d, level=SYS_NET_IPPROTO_IP, optname=%s, optval=*0x%x, optlen=%u)", s, static_cast<lv2_ip_option>(optname), optval, optlen);
+		break;
+	default:
+		sys_net.warning("sys_net_bnet_setsockopt(s=%d, level=0x%x, optname=0x%x, optval=*0x%x, optlen=%u)", s, level, optname, optval, optlen);
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/lv2/sys_net.cpp:3019]: Missing break

**Review by RipleyTom — APPROVED** [review]
Looks good.

**Comment by MarioSonic2987** [comment]
Fixes the crash present in recent versions of VSH/XMB (newer than 4.46).

**FW 4.87**
image

Windows version: 10.0.22000.376

**Comment by clienthax** [comment]
> Fixes the crash present in recent versions of VSH/XMB (newer than 4.46).
> 
> **FW 4.87** image
> 
> Windows version: 10.0.22000.376

Hm.. wondering if it's conflicting with something on my pc then as that's pretty close to my version

**Comment by arbrandes** [comment]
Did some bisecting, and this commit breaks SPU cache generation on Guitar Hero Live (BLES02180).  In other words, the game works on build 0.0.19-13134, but not 0.0.19-13135.  See [this thread](https://forums.rpcs3.net/thread-203621.html) in the forum.

**Comment by clienthax** [comment]
> Did some bisecting, and this commit breaks SPU cache generation on Guitar Hero Live (BLES02180). In other words, the game works on build 0.0.19-13134, but not 0.0.19-13135. See [this thread](https://forums.rpcs3.net/thread-203621.html) in the forum.

Yet there's no log, and this has nothing to do with spu

**Comment by arbrandes** [comment]
@clienthax, apologies for the missing log: I'm an RPCS3 newb.  Writing a regression report as we speak.  :)

(I have no hypotheses as to why this particular commit did that game in.  All I know is the previous build works, and this one doesn't.  🤷🏼‍♂️)

**Pull request merged**